### PR TITLE
feat(cli): added package.json as a source for cdk config

### PR DESCRIPTION
**Pull request by jdrydn**
Other Node packages such as [`mocha`](https://npm.im/mocha) & [`nyc`](https://npm.im/nyc) can optionally refer to config stored in `package.json`. This PR gets the CDK CLI to pull a `package.json` in the current working directory & read settings from either `aws/cdk` / `cdk` keys - I'm unsure which would be preferred so this PR allows for both.

e.g:
```
{
  "aws/cdk": {
    "app": "node ./cdk-stack.js"
  },
  "cdk": {
    "app": "node ./cdk-stack.js"
  }
}
```

I'm aware there should also be accompanying docs for this - happy to write them if someone can direct me to the repo/file to edit!

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by jdrydn** [comment]
> ### AWS CodeBuild CI Report
> * CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
> * Commit ID: [617f261](https://github.com/aws/aws-cdk/commit/617f26159d484afacc7ffc4d8022fd0b90693cad)
> * Result: FAILED
> * [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=279cb5f8-335b-47d4-b8bb-d5bc9577ffe5%2Fbuild.log) (available for 30 days)
> 
> _Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)_

> ```
> npm ERR! code ETARGET
> npm ERR! notarget No matching version found for @aws-cdk/aws-iot-actions@1.129.0.
> npm ERR! notarget In most cases you or one of your dependencies are requesting
> npm ERR! notarget a package version that doesn't exist.
> ```

File: packages/aws-cdk/lib/settings.ts [hunk 1]
```diff
@@ -284,6 +287,11 @@ export class Settings {
     });
   }
 
+  public static async fromPackageJson(fileName: string): Promise<Settings> {
+    const pkg = (await fs.pathExists(fileName)) ? await fs.readJson(fileName) : {};
+    return new Settings(pkg['aws/cdk'] || pkg.cdk);
```
Comment (rix0rrr) [inline packages/aws-cdk/lib/settings.ts:292]:
```suggestion
    return new Settings(pkg['awscdk'] ?? {});
```

**Review by rix0rrr — CHANGES_REQUESTED** [review]
Documentation should be added to the CLI README, and don't forget to add a test.

**Comment by rix0rrr** [comment]
I've mulled it over a little, and I'm not inclined to merge this. I don't see the value add, and merging this would reinforce the notion that CDK is a TS/JS-only tool, which we try our very best not to be.

**Pull request closed without merging**